framework/src/meshgenerators/PatternedMeshGenerator.C

```diff
@@ -88,35 +116,74 @@ PatternedMeshGenerator::generate()
                  type(),
                  " only works with inputs that are replicated.\n\n",
                  "Try running without distributed mesh.");
-    _meshes.push_back(std::move(mesh));
-  }
+    _meshes[i] = dynamic_pointer_cast<ReplicatedMesh>(mesh);
 
-  // Data structure that holds each row
-  _row_meshes.resize(_pattern.size());
+    const auto ids = MooseMeshUtils::getBoundaryIDs(*_meshes[i], boundary_names, false);
+    mooseAssert(ids.size() == boundary_names.size(),
+                "Unexpected number of ids returned for MooseMeshUtils::getBoundaryIDs");
 
-  // Getting the boundaries provided by the user
-  std::vector<BoundaryName> boundary_names = {getParam<BoundaryName>("left_boundary"),
-                                              getParam<BoundaryName>("right_boundary"),
-                                              getParam<BoundaryName>("top_boundary"),
-                                              getParam<BoundaryName>("bottom_boundary")};
+    // Keep track of used IDs so we can later find IDs that are unused across all meshes
+    used_boundary_ids.insert(ids.begin(), ids.end());
 
-  std::vector<boundary_id_type> ids =
-      MooseMeshUtils::getBoundaryIDs(*_meshes[0], boundary_names, true);
+    // Keep track of unique boundary ids seen in each input mesh
+    std::set<boundary_id_type> id_set;
+    // Keep track of indices of first instance of each unique boundary id
+    std::map<boundary_id_type, size_t> seen_bid_to_index_map;
+
+    size_t index = 0;
+    for (const auto side : make_range(4))
+    {
+      // Check if the boundary has been initialized
+      if (ids[side] == Moose::INVALID_BOUNDARY_ID)
+        paramError("inputs",
+                   "The '",
+                   boundary_param_names[side],
+                   "' parameter with value '",
+                   boundary_names[side],
+                   "' does not exist in input mesh '",
+                   _input_names[i],
+                   "'");
+
+      input_bids[i][side] = ids[side];
+
+      // We only do this when i == 0 because all input meshes should have the
+      // same index map. Allowing different index maps for different input
+      // meshes results in undefined behaviour when stitching
+      if (i == 0)
+      {
+        if (!id_set.count(ids[side]))
```

Comment:
you dont need id_set.
You're storing this information of "I have seen this before" in another vector (input_bids_unique) and a map seen_bid_to_index_map

Reply:
How do you recommend I replace the
`if (!id_set.count(ids[side]))`
statements? Use std::find on input_bids_unique and query that result instead?

Reply:
yeah there's a count on vectors and probably just find on the map